src/SEBT.Portal.Infrastructure/Services/SmtpClientService.cs

```diff
@@ -0,0 +1,43 @@
+using System.Net.Mail;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using SEBT.Portal.Core.AppSettings;
+using SEBT.Portal.Core.Services;
+
+namespace SEBT.Portal.Infrastructure.Services
+{
+    /// <summary>
+    /// Service for sending emails using SMTP protocol.
+    /// </summary>
+    /// <remarks>
+    /// This service wraps the <see cref="SmtpClient"/> to provide email sending capabilities
+    /// with configuration from <see cref="SmtpClientSettings"/>.
+    /// </remarks>
+    /// <param name="optionsMonitor">Options monitor for SMTP client settings.</param>
+    /// <param name="logger">Logger instance for logging email operations.</param>
+    public class SmtpClientService(IOptionsMonitor<SmtpClientSettings> optionsMonitor, ILogger<SmtpClientService> logger)
+        : ISmtpClientService
+    {
+        private readonly SmtpClientSettings smtpClientSettings = optionsMonitor.CurrentValue;
+
+        public async Task SendEmailAsync(MailMessage message)
+        {
+            // Configure the SMTP client
+            using var smtpClient = new SmtpClient(smtpClientSettings.SmtpServer, smtpClientSettings.SmtpPort)
```

Comment:
I think the native SmtpClient class currently discouraged from use from Microsoft themselves in favor of alternative libraries like `Mailkit`.  

https://learn.microsoft.com/en-us/dotnet/api/system.net.mail.smtpclient?view=net-10.0#remarks